Readability.js

```diff
@@ -1198,62 +1198,66 @@ Readability.prototype = {
 
-    // Match Facebook's Open Graph title & description properties.
-    var propertyPattern = /^\s*og\s*:\s*(description|title)\s*$/i;
+    // name is a single value
+    var namePattern = /^\s*(?:(dc|dcterm|og|twitter)\s*[\.:]\s*)?(author|creator|description|title)\s*$/gi;
 
     // Find description tags.
     this._forEachNode(metaElements, function(element) {
       var elementName = element.getAttribute("name");
       var elementProperty = element.getAttribute("property");
+      var content = element.getAttribute("content");
+      var matches = null;
+      var name = null;
 
-      if ([elementName, elementProperty].indexOf("author") !== -1) {
-        metadata.byline = element.getAttribute("content");
-        return;
+      if (elementProperty) {
+        matches = elementProperty.match(propertyPattern);
+        if (matches) {
+          for (var i = matches.length - 1; i >= 0; i--) {
+            // Convert to lowercase, and remove any whitespace
+            // so we can match below.
+            name = matches[i].toLowerCase().replace(/\s/g, "");
+            // multiple authors
+            values[name] = content.trim();
+          }
+        }
       }
-
-      var name = null;
-      if (namePattern.test(elementName)) {
+      if (!matches && namePattern.test(elementName)) {
         name = elementName;
-      } else if (propertyPattern.test(elementProperty)) {
-        name = elementProperty;
-      }
-
-      if (name) {
-        var content = element.getAttribute("content");
         if (content) {
-          // Convert to lowercase and remove any whitespace
-          // so we can match below.
-          name = name.toLowerCase().replace(/\s/g, "");
+          // Convert to lowercase, remove any whitespace, and convert dots
+          // to colons so we can match below.
+          name = name.toLowerCase().replace(/\s/g, "").replace(/\./g, ":");
           values[name] = content.trim();
         }
       }
     });
 
-    if ("description" in values) {
-      metadata.excerpt = values["description"];
-    } else if ("og:description" in values) {
-      // Use facebook open graph description.
-      metadata.excerpt = values["og:description"];
-    } else if ("twitter:description" in values) {
-      // Use twitter cards description.
-      metadata.excerpt = values["twitter:description"];
-    }
+    // get title
+    metadata.title = values["dc:title"] ||
+                     values["dcterm:title"] ||
+                     values["og:title"] ||
+                     values["title"] ||
+                     values["twitter:title"];
 
-    metadata.title = this._getArticleTitle();
     if (!metadata.title) {
-      if ("og:title" in values) {
-        // Use facebook open graph title.
-        metadata.title = values["og:title"];
-      } else if ("twitter:title" in values) {
-        // Use twitter cards title.
-        metadata.title = values["twitter:title"];
-      }
+      metadata.title = this._getArticleTitle();
```

Comment:
There's quite a bit of logic in `_getArticleTitle` to strip out the prefixed/suffixed junk (usually publication name or something like that) that we're now seeing in some of the metadata things. One thing we could potentially do is run that logic against the metadata as well. Doesn't need to happen here (though I wouldn't object if you did it here either) but might be nice as a separate PR once this lands, if you're interested in tidying this up further.

Reply:
This is only a problem with the tests. See [this comment](https://github.com/mozilla/readability/pull/478#issuecomment-415156928). `_getArticleTitle` only really works for English-language content and web developers have no control over it. It should only be used if no better data source available.